platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/utility/MerkleUtils.java

```diff
@@ -58,8 +65,12 @@ public static Hash rehashTree(final MerkleNode root) {
             final Future<Hash> future = MerkleCryptoFactory.getInstance().digestTreeAsync(root);
             try {
                 return future.get();
-            } catch (InterruptedException | ExecutionException e) {
+            } catch (InterruptedException e) {
+                logger.error(EXCEPTION.getMarker(), "Interrupted while waiting for tree hashing to complete", e);
                 Thread.currentThread().interrupt();
+            } catch (ExecutionException e) {
+                logger.error(EXCEPTION.getMarker(), "Async tree hashing failed", e);
+                throw new MerkleSynchronizationException(e);
```

Comment:
This exception type is intended for use by the reconnect algorithm (which is sometimes called "synchronization").

Reply:
Fair enough. I added an exception that is specific to this failure.